extensions/mongodb-client/runtime/src/main/java/io/quarkus/mongodb/runtime/AbstractMongoClientProducer.java

```diff
@@ -0,0 +1,384 @@
+package io.quarkus.mongodb.runtime;
+
+import static com.mongodb.AuthenticationMechanism.GSSAPI;
+import static com.mongodb.AuthenticationMechanism.MONGODB_X509;
+import static com.mongodb.AuthenticationMechanism.PLAIN;
+import static com.mongodb.AuthenticationMechanism.SCRAM_SHA_1;
+
+import java.lang.reflect.Constructor;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import javax.annotation.PreDestroy;
+
+import org.bson.codecs.configuration.CodecProvider;
+import org.bson.codecs.configuration.CodecRegistries;
+import org.bson.codecs.configuration.CodecRegistry;
+import org.bson.codecs.pojo.Conventions;
+import org.bson.codecs.pojo.PojoCodecProvider;
+import org.jboss.logging.Logger;
+
+import com.mongodb.AuthenticationMechanism;
+import com.mongodb.Block;
+import com.mongodb.ConnectionString;
+import com.mongodb.MongoClientSettings;
+import com.mongodb.MongoCredential;
+import com.mongodb.MongoException;
+import com.mongodb.ReadPreference;
+import com.mongodb.ServerAddress;
+import com.mongodb.WriteConcern;
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoClients;
+import com.mongodb.connection.ClusterConnectionMode;
+import com.mongodb.connection.ClusterSettings;
+import com.mongodb.connection.ConnectionPoolSettings;
+import com.mongodb.connection.ServerSettings;
+import com.mongodb.connection.SocketSettings;
+import com.mongodb.connection.SslSettings;
+
+import io.quarkus.mongodb.ReactiveMongoClient;
+import io.quarkus.mongodb.impl.ReactiveMongoClientImpl;
+
+public abstract class AbstractMongoClientProducer {
+
+    private static final Logger LOGGER = Logger.getLogger(AbstractMongoClientProducer.class.getName());
+    private static final Pattern COLON_PATTERN = Pattern.compile(":");
+    private MongodbConfig mongodbConfig;
+    //private boolean disableSslSupport = false;
```

Comment:
Can you please explain what is going on with this field? Why is it commented out?

Reply:
it was not supported on previous extention. I have just add support of it. But I do not know if it was a bug/missing feature or really wanted.

Reply:
anyway I have add the support.